kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/PaginatedScanSuite.scala

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.util.Optional
+
+import io.delta.kernel.PaginatedScan
+import io.delta.kernel.PaginatedScanFilesIterator
+import io.delta.kernel.data.FilteredColumnarBatch
+import io.delta.kernel.data.Row
+import io.delta.kernel.defaults.DeltaTableWriteSuiteBase
+import io.delta.kernel.defaults.engine.{DefaultEngine, DefaultJsonHandler, DefaultParquetHandler}
+import io.delta.kernel.defaults.test.AbstractTableManagerAdapter
+import io.delta.kernel.defaults.utils.{ExpressionTestUtils, TestUtils}
+import io.delta.kernel.defaults.utils.TestUtilsWithTableManagerAPIs
+import io.delta.kernel.internal.replay.{PageToken, PaginatedScanFilesIteratorImpl}
+
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.scalatest.funsuite.AnyFunSuite
+import org.slf4j.{Logger, LoggerFactory}
+
+class PaginatedScanSuite extends AnyFunSuite with TestUtilsWithTableManagerAPIs
+    with ExpressionTestUtils with SQLHelper with DeltaTableWriteSuiteBase {
+
+  private val logger = LoggerFactory.getLogger(classOf[PaginatedScanSuite])
+  val tableManager: AbstractTableManagerAdapter = getTableManagerAdapter
+
+  /** Test case to check if the result of a single page request is expected */
+  case class SinglePageRequestTestCase(
+      pageSize: Int,
+      expFileCnt: Int,
+      expBatchCnt: Int,
+      expLogFile: String,
+      expRowIdx: Int)
+
+  /** Custom engine with customized batch size */
+  private val customEngine: DefaultEngine = {
+    val BATCH_SIZE = 5
+    val hadoopConf = new org.apache.hadoop.conf.Configuration()
+    hadoopConf.set("delta.kernel.default.json.reader.batch-size", BATCH_SIZE.toString)
```

Comment:
worth mentioning in a comment that we must NOT change this batch size -- it is hardcoded and impacts every single test!